exporter/fairshare.go

```diff
@@ -0,0 +1,156 @@
+package exporter
+
+import (
+	"fmt"
+	"log/slog"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type FairShareMetric struct {
+	Account   string
+	FairShare float64
+}
+
+type FairShareFetcher struct {
+	scraper      SlurmByteScraper
+	errorCounter prometheus.Counter
+	cache        *AtomicThrottledCache[FairShareMetric]
+}
+
+func (fsf *FairShareFetcher) parseFairShare(output string) ([]FairShareMetric, error) {
+	var metrics []FairShareMetric
+	lines := strings.Split(output, "\n")
+
+	for _, line := range lines {
+		// Check if line contains data
+		if !strings.Contains(line, "|") {
+			continue
+		}
+
+		parts := strings.Split(line, "|")
+		if len(parts) < 2 {
+			continue
+		}
+
+		account := strings.TrimSpace(parts[0])
+		if account == "" {
+			continue
+		}
+
+		fairshareStr := strings.TrimSpace(parts[1])
+		// Skip empty values and "inf" values
+		if fairshareStr == "" || fairshareStr == "inf" {
+			continue
+		}
+
+		fairshare, err := strconv.ParseFloat(fairshareStr, 64)
+		if err != nil {
+			slog.Warn("Failed to parse fairshare value", "account", account, "value", fairshareStr, "err", err)
+			continue
+		}
+
+		metrics = append(metrics, FairShareMetric{
+			Account:   account,
+			FairShare: fairshare,
+		})
+	}
+
+	return metrics, nil
+}
+
+func (fsf *FairShareFetcher) fetchFromCli() ([]FairShareMetric, error) {
+	output, err := fsf.scraper.FetchRawBytes()
+	if err != nil {
+		fsf.errorCounter.Inc()
+		slog.Error(fmt.Sprintf("failed to scrape fairshare metrics with %q", err))
+		return nil, err
+	}
+
+	return fsf.parseFairShare(string(output))
+}
+
+func (fsf *FairShareFetcher) FetchMetrics() ([]FairShareMetric, error) {
+	return fsf.cache.FetchOrThrottle(fsf.fetchFromCli)
+}
+
+func (fsf *FairShareFetcher) ScrapeError() prometheus.Counter {
+	return fsf.errorCounter
+}
+
+func (fsf *FairShareFetcher) ScrapeDuration() time.Duration {
+	return fsf.scraper.Duration()
+}
+
+type FairShareCollector struct {
+	fetcher                 SlurmMetricFetcher[FairShareMetric]
+	accountFairShare        *prometheus.Desc
+	fairshareScrapeDuration *prometheus.Desc
+	fairshareScrapeError    prometheus.Counter
+}
+
+func NewFairShareCollector(config *Config) *FairShareCollector {
+	return &FairShareCollector{
+		fetcher: &FairShareFetcher{
+			scraper: NewCliScraper("sshare", "-n", "-P", "-o", "account,levelfs"),
```

Comment:
Let's parameterize the config to pass in a custom cmd if desired. For example, say a user has wrote their own slurm plugin that outputs compatible information or want to explicitly point to a different binary. Please add fairshare `string[]` [here](https://github.com/rivosinc/prometheus-slurm-exporter/blob/main/exporter/server.go#L25), add the default in the [constructor](https://github.com/rivosinc/prometheus-slurm-exporter/blob/main/exporter/server.go#L86), and invoke it as such:



```suggestion
            cliOpts := config.cliOpts
            
			scraper: NewCliScraper(cliOpts.newOpt...),
```